Message-Id: <200307311239.h6VCdNe27056@r-rr.iij4u.or.jp>
To: dank@kegel.com
Cc: kkojima@rr.iij4u.or.jp
Subject: Re: Writing PIC version of __udivsi3_i4? (was: Re: "make tests" fails
 with "libc.so: text relocations used" in glibc-2.3.2 for sh4)
In-Reply-To: Your message of "Wed, 30 Jul 2003 21:55:54 -0700"
	<3F28A15A.5010400@kegel.com>
References: <3F28A15A.5010400@kegel.com>
Date: Thu, 31 Jul 2003 21:47:38 +0900
From: kaz Kojima <kkojima@rr.iij4u.or.jp>

Dan Kegel <dank@kegel.com> wrote:
>> I'll give binutils-2.14.90.0.5 a shot, then. 
> 
> No joy - same problem.  Guess I'll take this up on binutils and/or libc-alpha.

Please try the patch below, though it's a diff against the current
binutils CVS.

Regards,
	kaz
--
diff -u3p ORIG/src/bfd/elf32-sh.c LOCAL/src/bfd/elf32-sh.c
--- src/bfd/elf32-sh.c.old	Fri Jul 25 14:46:58 2003
+++ src/bfd/elf32-sh.c	Thu Jul 31 21:09:38 2003
@@ -6497,8 +6497,10 @@ sh_elf_check_relocs (bfd *abfd, struct b
 							  sreloc, 2))
 			return FALSE;
 		    }
+#if 0
 		  if (sec->flags & SEC_READONLY)
 		    info->flags |= DF_TEXTREL;
+#endif
 		  elf_section_data (sec)->sreloc = sreloc;
 		}
 


